diff --git a/zenoh/src/api/timestamp_stack.rs b/zenoh/src/api/timestamp_stack.rs
--- a/zenoh/src/api/timestamp_stack.rs
+++ b/zenoh/src/api/timestamp_stack.rs
@@ -432,11 +432,6 @@
             // Skip writing empty Vec
             return;
         }
-        if ts_stack.ts_stack.stack.len() >= zenoh_protocol::network::timestamp_stack::MAX_STACK_SIZE
-        {
-            // Avoid producing an invalid frame that will be rejected by the codec decoder.
-            return;
-        }
         ts_stack.ts_stack.stack.push(Interception {
             flags: point
                 | if is_custom {
